### ISSUE TITLE
Oh god here goes NSIS again

### ISSUE THREAD

**Issue by fasterthanlime** [itchio/itch#2327]
Follow-up to #2326, I'm trying to figure out what recent NSIS versions do.

For electron-builder apps, there's 3 kinds of targets.

There's **normal** (so called because this is what the itch app expects from NSIS installers):

```javascript
{
  "win": {
    "target": "nsis"
  },
  "nsis": {
    "oneClick": false
  }
}
```

There's **oneclick**:

```javascript
{
  "win": {
    "target": "nsis"
  },
  "nsis": {
    "oneClick": true // the default for "nsis" target
  }
}
```

And there's **portable**:

```javascript
{
  "win": {
    "target": "portable"
  },
}
```

**Comment by fasterthanlime** [comment]
Attached to this comment are all 3 variants generated from https://github.com/fasterthanlime/electron-builder-sample (a private repo).

I've used proton-native for size, but electron-builder's result is the same with proton-native apps and electron apps, so that doesn't matter.

  * 📦 [nsis-normal.zip](https://github.com/itchio/itch/files/3646896/nsis-normal.zip)
  * 📦 [nsis-oneclick.zip](https://github.com/itchio/itch/files/3646897/nsis-oneclick.zip)
  * 📦 [nsis-portable.zip](https://github.com/itchio/itch/files/3646898/nsis-portable.zip)

**Comment by fasterthanlime** [comment]
Unfortunately, all three variants yield the same "spell" (result of file(1) Magic rules):

![image](https://user-images.githubusercontent.com/7998310/65508165-e9958e80-decf-11e9-96c7-725262824e47.png)

**Comment by fasterthanlime** [comment]
They also have the exact same manifest:

![image](https://user-images.githubusercontent.com/7998310/65508236-0b8f1100-ded0-11e9-8695-afc08d93e956.png)

**Comment by fasterthanlime** [comment]
The "normal" variant accepts silent installation:

![image](https://user-images.githubusercontent.com/7998310/65508483-9d971980-ded0-11e9-827e-503fa1db57e9.png)

So does the "oneclick" variant:

![image](https://user-images.githubusercontent.com/7998310/65508539-bd2e4200-ded0-11e9-9b68-d789878aabec.png)

But the "portable" variant does not:

![image](https://user-images.githubusercontent.com/7998310/65508623-e5b63c00-ded0-11e9-841c-565adf7dd07f.png)

**Comment by fasterthanlime** [comment]
As expected, 7-zip (but not lib7zip for some reason) is able to unpack the installer, and inside there's a .7z file, which, again, 7-zip can extract, and *in that case* it does contain the full app, so that seems ideal, right?

Wrong! Because NSIS is scriptable and the script could contain *another* unpacked if they wanted, so even unpacking twice (assuming we find some way to make it work with lib7zip) is not a solution here.

I found that out 5 months ago the hard way: https://github.com/itchio/itch/issues/2281 - apparently 5 months is enough to forget.

TL;DR - wontfix, NSIS installers are sort of a lost cause.
